src/event/v03/attributes.rs

```diff
@@ -17,6 +17,60 @@ pub struct Attributes {
     pub(crate) time: Option<DateTime<Utc>>,
 }
 
+impl<'a> IntoIterator for &'a Attributes {
+    type Item = (&'a str, AttributeValue<'a>);
+    type IntoIter = AttributesIntoIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        AttributesIntoIterator {
+            attributes: self,
+            index: 0,
+        }
+    }
+}
+
+pub struct AttributesIntoIterator<'a> {
+    attributes: &'a Attributes,
+    index: usize,
+}
+
+impl<'a> Iterator for AttributesIntoIterator<'a> {
+    type Item = (&'a str, AttributeValue<'a>);
+    fn next(&mut self) -> Option<Self::Item> {
+        let result = match self.index {
+            0 => Some(("id", AttributeValue::String(&self.attributes.id))),
+            1 => Some(("ty", AttributeValue::String(&self.attributes.ty))),
+            2 => Some(("source", AttributeValue::URI(&self.attributes.source))),
```

Comment:
For the spec this is a URIRef: https://github.com/cloudevents/spec/blob/v1.0/spec.md#source-1